core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -824,6 +824,8 @@ private[spark] class TaskSetManager(
         if (fetchFailed.bmAddress != null) {
           healthTracker.foreach(_.updateExcludedForFetchFailure(
             fetchFailed.bmAddress.host, fetchFailed.bmAddress.executorId))
+          // Remove fetchFailed host in the shuffle push merger list for push based shuffle
+          env.blockManager.master.removeShufflePushMergerLocation(fetchFailed.bmAddress.host)
```

Comment:
hmm..I'd prefer to handle this inside DAGScheduler...DAGScheduler handles fetch failure too and it also checks whether external shuffle service is used in case of the failure, see: 
https://github.com/apache/spark/blob/72ad9dcd5d484a8dd64c08889de85ef9de2a6077/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L1826